### PR TITLE
FAU Development

### PR DESCRIPTION
**Pull request by yushan87**
This pull request contains the new alternative version for the tutor developed by the FAU folks. Since it is such a big change, I am requesting reviews from @jzhai2016, @SeaGuy, and @timschwab so that we can ensure everything works properly before I merge this.

@timschwab Wait until we talk in person before you do anything with this pull request. 
@jzhai2016 @SeaGuy If we discover bugs or find new changes, you can change it on your end and push to this branch. The pull request will update and include your new commits!

### CODE REVIEW

File: backend/sendData.js [hunk 1]
```diff
@@ -13,5 +13,5 @@ function sendData() {
     data.correct = approved;
     data.points = "0";
 
-    $.post("https://resolve.cs.clemson.edu/teaching/bydesign", JSON.stringify(data));
+    $.post("https://resolve.cs.clemson.edu/testing/bydesign", JSON.stringify(data));
```
Comment (yushan87) [inline backend/sendData.js]: Can someone switch this back to `teaching`?

**Comment by yushan87** [comment]
I will fix the `hound` issues tomorrow.

**Comment by yushan87** [comment]
Took a while, but I am done with all the linting errors. Hopefully I didn't break anything...

**Comment by timschwab** [comment]
It turned out that when you guys look for VCs, you are assuming that you are in the FAU section. So when our sections run the code, it can never find any VCs, and it always approves the code, even if it is wrong. I fixed this by separating our two sections by testing whether the `currentLesson` had a `base`. But in doing this, I seem to have broken the FAU code. I looked into a bit, but was unsure what I had changed. Perhaps you guys can find it easier than me.

**Comment by jzhai2016** [comment]
Hi Tim,

Sorry for the late response! Bill and I have been busy with a couple of deadlines. I’ll take a look and get back to you.

Best,
Jiannan

**Comment by jzhai2016** [comment]
Hi Tim,

I fixed the bug. Maybe it’s a typo. You added a “,” to an if statement I wrote. Removing it solved the problem.

Best,
Jiannan

From: Jiannan Zhai
Sent: Friday, March 31, 2017 4:07 PM
To: 'ClemsonRSRG/beginToReason' <reply@reply.github.com>
Subject: RE: [ClemsonRSRG/beginToReason] FAU Development (#43)

Hi Tim,

Sorry for the late response! Bill and I have been busy with a couple of deadlines. I’ll take a look and get back to you.

Best,
Jiannan

**Comment by timschwab** [comment]
Oh good, I'm glad it was a simple fix. I changed the link to the server back to the original URL, switching from `wss://resolve.cs.clemson.edu/testing/Compiler` to `wss://resolve.cs.clemson.edu/teaching/Compiler`. It looks good to me, and I will merge it. Thank you for your work!

**Pull request merged**